sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereQuery.cs

```diff
@@ -117,16 +203,24 @@ public override AdtQueryBuilder Build()
         /// <inheritdoc/>
         public override string Stringify()
         {
-            // TODO -- make this cleaner? design problem?
             if (_clauses.Any())
             {
-                // TODO -- if appending more strings turn into a stringbuilder
-                string whereComponents = $"{QueryConstants.Where} ";
-
-                // TODO -- turn condition into a string
-                // add where arguments (conditions)
-
-                return whereComponents;
+                // Where keyword only needs to be appened one time, happends outside of loop
+                var whereComponents = new StringBuilder();
+                whereComponents.Append($"{QueryConstants.Where} ");
+
+                // Parse each Where conditional statement
+                for (int i = 0; i < _clauses.Count - 1; i++)
```

Comment:
I think you are one short!
why not use foreach?

Reply:
initially had it set up to use `i` outside of the loop, forgot to change it back. Will change to foreach. I append the last argument outside of the loop since it's an edge case where we don't want an extra `And()` at the end, hence, once short.